File: client/src/app/model/user-dto.ts
import {Address} from './address';
import {AddressDTO} from './address-dto.model';

export class UserDTO {
  public firstName!: string;
  public email!: string;
  public surname!: string;
  public password!: string;
  public phoneNumber!: string;
  public address: AddressDTO = ({} as any);
}

File: isa2/demo/src/main/java/isa2/demo/DTO/UserDTO.java
package isa2.demo.DTO;

import isa2.demo.Model.Address;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UserDTO {
    private String firstName;
    private String surname;
    private String password;
    private String email;
    private String phoneNumber;
    private AddressDTO address;
    private Boolean firstLogin;
}
